### DISCUSSION TITLE
@types/react-dom: React.CreatePortal should allow ShadowDom type as container

### DISCUSSION THREAD

**Discussion by rushil-patel** [DefinitelyTyped/DefinitelyTyped#53619] [Issues with a @types package]
`React.createPortal` is typed to expect an `Element` to be passed as the container. It's typed [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-dom/index.d.ts#L28) as: 
```
createPortal(children: ReactNode, container: Element, key?: null | string): ReactPortal;
```
However according to the createPortal [implementation](https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOM.js#L116) a valid container can be an Element, a Document, a Document Fragment, and sometimes a Comment node. 

I ran into this issue trying to use a ShadowRoot which extends DocumentFragment as a container. It would seem that Element is no longer the appropriate type. The type definition should be updated to support Document and Document Fragments
```js

const ref = useRef<HTMLDivElement>(null);
const shadowRootNode: ShadowRoot = ref.current.attachShadow({
    mode
});

// type error: Argument of type 'ShadowRoot' is not assignable to parameter of type 'Element'.
ReactDOM.createPortal(children, shadowRootNode);
```

**Comment by davbrito** [comment]
I currently have to cast the type explicitliy (`shadowRoot as unknown as Element`) for this to work.
Seems like no much people are interested into this.
